libusb/os/windows_winusb.c

```diff
@@ -1237,7 +1241,12 @@ static bool get_dev_port_number(HDEVINFO dev_info, SP_DEVINFO_DATA *dev_info_dat
 			NULL, (PBYTE)buffer, sizeof(buffer), NULL)) {
 		// Check for the required format.
 		if (strncmp(buffer, "Port_#", 6) == 0) {
-			*port_nr = atoi(buffer + 6);
+			start = buffer + 6;
+			port = strtoll(start, &end, 10);
+			if (port < 0 || port > 4294967295 || end == start || *end != '\0') {
```

Comment:
Why not MAX_INT or something here? And can there be a possibility of such high port numbers anyway? From the above comment I wouldn't expect anything above 0xffff.

Reply:
>Why not MAX_INT or something here? 

Simply because I couldn't find such a constant for the maximum `DWORD`. Do you know one? The point of this line is to make 100% sure the cast a couple of lines later is safe:

```
*port_nr = (DWORD)port;
```

>And can there be a possibility of such high port numbers anyway? 

`strtoll` returns `long long` so I was being maximally paranoid.

>From the above comment I wouldn't expect anything above 0xffff.

In fact the comment is "The string *may* have a format similar to..." with "may" emphasized, as if in doubt. If `pSetupDiGetDeviceRegistryPropertyA` puts some crazyness in the buffer, who knows...

But if you're sure 0xFFFF is the max valid, we could tighten the check.

Again, I'm only trying to get rid of old `atoi` here. The old code was not limiting to 0xFFFF.

Reply:
> Simply because I couldn't find such a constant for the maximum `DWORD`. Do you know one?

DWORD is typedef unsigned long, so ULONG_MAX.

Reply:
Done.